watertap/unit_models/zero_order/coag_and_floc_zo.py

```diff
@@ -0,0 +1,226 @@
+        self._fixed_perf_vars.append(self.G_rapid_mix)
+        self._fixed_perf_vars.append(self.G_floc)
+
+        self._perf_var_dict["Alum Dosage (mg/L)"] = self.alum_dose
+        self._perf_var_dict["Polymer Dosage (mg/L)"] = self.polymer_dose
+        self._perf_var_dict["Alum Flow (kg/s)"] = self.chemical_flow_mass[0, "alum"]
+        self._perf_var_dict["Polymer Flow (kg/s)"] = self.chemical_flow_mass[0, "polymer"]
+        self._perf_var_dict["Rapid Mix Basin Volume (m^3)"] = self.rapid_mix_basin_vol
+        self._perf_var_dict["Floc Basin Volume (m^3)"] = self.floc_basin_vol
+        self._perf_var_dict["Rapid Mix Retention Time (s)"] = self.rapid_mix_retention_time
+        self._perf_var_dict["Floc Retention Time (min)"] = self.floc_retention_time
+        self._perf_var_dict["Number of Floc Injection Processes"] = self.num_floc_injection_processes
+        self._perf_var_dict["Number of Floc Processes"] = self.num_floc_processes
+        self._perf_var_dict["Number of Rapid Mixers"] = self.num_rapid_mixers
+        self._perf_var_dict["Number of Coagulation Processes"] = self.num_coag_processes
+        self._perf_var_dict["Number of Floc Mixers"] = self.num_floc_mixers
+        self._perf_var_dict["Number of Rapid Mix Processes"] = self.num_rapid_mix_processes
+        self._perf_var_dict["Rapid Mix Velocity Gradient (1/s)"] = self.G_rapid_mix
+        self._perf_var_dict["Floc Velocity Gradient (1/s)"] = self.G_floc
+        self._perf_var_dict["Rapid Mix Power (kW)"] = self.power_rapid_mix
+        self._perf_var_dict["Floc Power (kW)"] = self.power_floc
+        self._perf_var_dict["Electricity intensity per Feed Flow Rate (kWh/m^3)"] = self.energy_electric_flow_vol_inlet
+
+        def rule_rapid_mix_basin_vol(blk):
+            return (blk.rapid_mix_basin_vol ==
+                    blk.properties[0].flow_vol * blk.rapid_mix_retention_time[0])
+        self.rapid_mix_basin_vol_constraint = Constraint(rule=rule_rapid_mix_basin_vol)
+
+        def rule_floc_basin_vol(blk):
+            return (blk.floc_basin_vol ==
+                    pyunits.convert(blk.properties[0].flow_vol * blk.floc_retention_time[0],
+                                    to_units=pyunits.m**3))
+        self.floc_basin_vol_constraint = Constraint(rule=rule_floc_basin_vol)
+
+        def rule_chem_flow(blk, t, j):
+            if j == 'alum':
+                chemical_dosage = blk.alum_dose[t]
+            elif j == 'polymer':
+                chemical_dosage = blk.polymer_dose[t]
+            return (blk.chemical_flow_mass[t, j] ==
+                    pyunits.convert(chemical_dosage * blk.properties[t].flow_vol,
+                                    to_units=pyunits.kg/pyunits.s))
+        self.chemical_flow_constraint = Constraint(self.flowsheet().time,
+                                                   ["alum", "polymer"],
+                                                   rule=rule_chem_flow)
+
+        def rule_anionic_polymer_dose(blk, t):
```

Comment:
I wonder if it would be best if these were Vars and Constraints instead of Expressions, just to keep things more consistent.

Reply:
I can do that.